pyodk/endpoints/submissions.py

```diff
@@ -159,12 +153,165 @@ def get_table(
         except PyODKError as err:
             log.error(err, exc_info=True)
             raise err
-        else:
-            response = self.session.get_200_or_error(
-                url=self.urls.get_table.format(
-                    project_id=pid, form_id=fid, table_name=table
-                ),
-                logger=log,
-                params=params,
+
+        response = self.session.response_or_error(
+            method="GET",
+            url=self.urls.get_table.format(project_id=pid, form_id=fid, table_name=table),
+            logger=log,
+            params=params,
+        )
+        return response.json()
+
+    def post(
+        self,
+        xml: str,
+        form_id: Optional[str] = None,
+        project_id: Optional[int] = None,
+        device_id: Optional[str] = None,
+    ) -> Submission:
+        """
+        Create a Submission.
+
+        Example submission XML structure:
+
+        <data id="my_form" version="v1">
+          <meta>
+            <instanceID>uuid:85cb9aff-005e-4edd-9739-dc9c1a829c44</instanceID>
+          </meta>
+          <name>Alice</name>
+          <age>36</age>
+        </data>
+
+        :param xml: The submission XML.
+        :param form_id: The xmlFormId of the Form being referenced.
+        :param project_id: The id of the project this form belongs to.
+        :param device_id: An optional deviceID associated with the submission.
+        """
+        try:
+            pid = pv.validate_project_id(project_id, self.default_project_id)
+            fid = pv.validate_form_id(form_id, self.default_form_id)
+            params = {}
+            if device_id is not None:
+                params["deviceID"] = device_id
+        except PyODKError as err:
+            log.error(err, exc_info=True)
+            raise err
+
+        response = self.session.response_or_error(
+            method="POST",
+            url=self.urls.post.format(project_id=pid, form_id=fid),
+            logger=log,
+            headers={"Content-Type": "application/xml"},
+            params=params,
+            data=xml,
+        )
+        data = response.json()
+        return Submission(**data)
+
+    def put(
```

Comment:
How about making this one "private" (prepend with _ and don't document). In general we want to encourage users to make a comment with an edit. They can omit the `comment` parameter to `edit` if they really want. (I had suggested making it required! But I'm ok with it being optional)

Here's the Central frontend UI if you make an edit without a comment:
<img width="624" alt="Screen Shot 2022-10-11 at 2 22 02 PM" src="https://user-images.githubusercontent.com/967540/195200854-a0d6f781-1967-4e00-9e0a-1a3d5f3f7707.png">

We would have made it required if we could have but because of the way we communicate with Enketo we can't.

Reply:
_put marked private